common/src/main/java/org/openmrs/analytics/FhirStoreUtil.java

```diff
@@ -89,27 +89,36 @@ public void uploadResourceToCloud(String resourceType, String resourceId, Resour
 	}
 	
 	// TODO: merge the two versions of this method to remove redundant resource info
-	public void uploadResourceToCloud(String resourceType, String resourceId, String fhirJson) {
-		uploadResourceToCloud(resourceType, resourceId, (Resource) fhirContext.newJsonParser().parseResource(fhirJson));
+	public void uploadResourceToCloud(String resourceType, String resourceId, String fhirJson) throws Exception {
+		if (httpUtil.checkServerConnection("http://google.com")) {
```

Comment:
I don't think it is a good idea to rely on checking an external service (even if it is `google.com`). The main idea of issue #32 was to deal with intermittent  issues, e.g., it fails to upload a resource once but succeeds after a few seconds. Let's implement that logic.

Reply:
sure makes sense